Backpack-Common/Carousel/Classes/BPKInternalCarousel.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public protocol BPKInternalCarouselDelegate: AnyObject {
+    func onImageChange(index: Int)
+}
+
+public class BPKInternalCarousel: UIView {
```

Comment:
Moving the existing UIKit carousel to Backpack-Common, renaming it to `Internal`.
It still needs to be public, as this is essentially a different module, that needs to be consumed by Backpack and Backpack-SwiftUI.